PR 15230 introduces test failure for particle_argon

git bisect identifies 7b1ee5cf13dc4f130b556534bd4c8dd78ceec6c6 as introducing this failure, which blocks testing of #15369.
```
lilith[402]$ git reset --hard 7b1ee5cf13dc4f130b556534bd4c8dd78ceec6c6
HEAD is now at 7b1ee5cf13 tests: CONFIG_TEST_USERSPACE now off by default
lilith[403]$ rm -rf argon ; west build -d argon -b particle_argon tests/drivers/spi/spi_loopback/
source directory: /mnt/devel/external/zp/zephyr/tests/drivers/spi/spi_loopback
build directory: /mnt/devel/external/zp/zephyr/argon (created)
BOARD: particle_argon
Zephyr version: 1.14.0
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.7", minimum required is "3.4") 
-- Selected BOARD particle_argon
-- Found west: /home/pab/.local/bin/west (found suitable version "0.5.7", minimum required is "0.5.6")
-- Loading /mnt/devel/external/zp/zephyr/boards/arm/particle_argon/particle_argon.dts as base
-- Overlaying /mnt/devel/external/zp/zephyr/dts/common/common.dts
particle_argon.dts.pre.tmp:312.23-315.5: Warning (simple_bus_reg): /soc/virtualcom: missing or empty reg/ranges property
Parsing Kconfig tree in /mnt/devel/external/zp/zephyr/tests/drivers/spi/spi_loopback/Kconfig
Loading /mnt/devel/external/zp/zephyr/boards/arm/particle_argon/particle_argon_defconfig as base
Merging /mnt/devel/external/zp/zephyr/tests/drivers/spi/spi_loopback/prj.conf
Configuration written to '/mnt/devel/external/zp/zephyr/argon/zephyr/.config'
-- Cache files will be written to: /home/pab/.cache/zephyr
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/local/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Performing Test toolchain_is_ok
-- Performing Test toolchain_is_ok - Success
Including module: tinycbor in path: /mnt/devel/external/zp/modules/lib/tinycbor
CMake Warning at ../../../../CMakeLists.txt:1468 (message):
  

        ------------------------------------------------------------
        --- WARNING:  __ASSERT() statements are globally ENABLED ---
        --- The kernel will run more slowly and use more memory  ---
        ------------------------------------------------------------


-- Configuring done
CMake Error at ../../../../cmake/extensions.cmake:357 (add_library):
  No SOURCES given to target: drivers__spi
Call Stack (most recent call first):
  ../../../../cmake/extensions.cmake:334 (zephyr_library_named)
  ../../../../drivers/spi/CMakeLists.txt:1 (zephyr_library)


-- Build files have been written to: /mnt/devel/external/zp/zephyr/argon
ERROR: command exited with status 1: /usr/local/bin/cmake -B/mnt/devel/external/zp/zephyr/argon -H/mnt/devel/external/zp/zephyr/tests/drivers/spi/spi_loopback -GNinja -DBOARD=particle_argon
run as "west -v build -d argon -b particle_argon tests/drivers/spi/spi_loopback/" for a stack trace
```

--- a/boards/arm/particle_argon/Kconfig.defconfig
+++ b/boards/arm/particle_argon/Kconfig.defconfig
@@ -23,6 +23,13 @@ config I2C_0
 
 endif # I2C
 
+if SPI
+
+config SPI_2
+       default y
+
+endif # SPI
+
 if USB
 
 config USB_NRF52840

--- a/boards/arm/particle_boron/Kconfig.defconfig
+++ b/boards/arm/particle_boron/Kconfig.defconfig
@@ -23,6 +23,13 @@ config I2C_0
 
 endif # I2C
 
+if SPI
+
+config SPI_2
+       default y
+
+endif # SPI
+
 if USB
 
 config USB_NRF52840

--- a/boards/arm/particle_xenon/Kconfig.defconfig
+++ b/boards/arm/particle_xenon/Kconfig.defconfig
@@ -24,6 +24,13 @@ config I2C_0
 
 endif # I2C
 
+if SPI
+
+config SPI_2
+       default y
+
+endif # SPI
+
 if USB
 
 config USB_NRF52840

